### PR TITLE
Add support for Google Pub/Sub as transport broker

### PR DESCRIPTION
**Pull request by haimjether**
Add support for GCP Pub/Sub 

* Add tests
* Add docs

Fix: https://github.com/celery/kombu/issues/705
Fix: https://github.com/celery/celery/issues/3984

### CODE REVIEW

**Review by Nusnus — CHANGES_REQUESTED** [review]
Please fix CI failures 🙏

**Review by Nusnus — CHANGES_REQUESTED** [review]
Also please change back to draft until the CI fully passes.
Feel free to ping me when it's ready for review!

**Comment by haimjether** [comment]
> Please fix CI failures 🙏

@Nusnus , I can't re-trigger those without maintainer approval :(

**Comment by haimjether** [comment]
> Also please change back to draft until the CI fully passes. Feel free to ping me when it's ready for review!

ping :)

**Comment by haimjether** [comment]
My bad - I have last tiny nit,
would you mind if I push it  ?

**Comment by Nusnus** [comment]
> My bad - I have last tiny nit,
> would you mind if I push it  ? 

Go for it 💯

File: tox.ini
Comment (Nusnus) [inline tox.ini:28]: Cool!

**Review by Nusnus — APPROVED** [review]
Thank you for the great contribution!
Amazing work, end to end.

Well done 💯👏

**Comment by Nusnus** [comment]
I'm going to release a new version soon.
Anything else required on the Kombu side? @haimjether

**Comment by haimjether** [comment]
@Nusnus ,
kombu's off the hook (for now) :) 

However this one https://github.com/celery/celery/pull/9351 
pretty mandatory for Flower to work correctly and is at Celery playground.

**Pull request merged**
